src/Query/Filter/TextField.ts

```diff
@@ -12,49 +12,75 @@ import { FilterOrErrorMessage } from './Filter';
+
+                    if (filterMethod.includes('include')) {
+                        matcher = new SubstringMatcher(searchString);
+                    } else if (filterMethod.includes('match')) {
+                        matcher = RegexMatcher.validateAndConstruct(searchString);
+                        if (matcher === null) {
+                            let msg = `cannot parse regular expression for ${fieldName}`;
+                            if (!new RegExp('/(.*)/i?').test(searchString)) {
+                                msg += '; check that it has leading and trailing slashes';
+                            }
+                            return FilterOrErrorMessage.fromError(msg);
+                        }
+                    }
+
+                    if (matcher) {
+                        // should always be the case if the filterRegexp is correct
 
-        if (matcher === null) {
-            // It's likely this can now never be reached.
-            // Retained for safety, for now.
-            return FilterOrErrorMessage.fromError(`do not understand query filter (${this.fieldName()})`);
+                        // for fields with multiple values, use the appropriate filter
+                        const filter = fieldName.includes('/')
+                            ? (task: Task) => matcher!.matchesAnyOf(this.values(task))
+                            : (task: Task) => matcher!.matches(this.value(task));
+
+                        // Create the final Filter, that takes a task and tests
+                        // if it matches the string filtering rule represented by this object.
+                        return FilterOrErrorMessage.fromFilter((task: Task) =>
+                            TextField.maybeNegate(filter(task), filterMethod),
+                        );
+                    }
+                }
+            }
         }
 
-        // Finally, we can create the Filter, that takes a task
-        // and tests if it matches the string filtering rule
-        // represented by this object.
-        return FilterOrErrorMessage.fromFilter((task: Task) => {
-            return TextField.maybeNegate(matcher!.matches(this.value(task)), filterMethod);
-        });
+        return FilterOrErrorMessage.fromError(`do not understand query filter (${fieldName})`);
     }
 
     public static stringIncludesCaseInsensitive(haystack: string, needle: string): boolean {
         return SubstringMatcher.stringIncludesCaseInsensitive(haystack, needle);
     }
 
     protected filterRegexp(): RegExp {
-        return new RegExp(`^${this.fieldName()} (includes|does not include|regex matches|regex does not match) (.*)`);
+        let fieldName = this.fieldName();
+        let ops = 'includes|does not include|(?:regexp? )?(?:matches|does not match)';
+        if (fieldName.includes('/')) {
+            fieldName = fieldName.replace('/', '|');
+            // add plural form of operators, but to be tolerant and backward compatible,
```

Comment:
Yes, will definitely change the slash as discriminator thing.

Reply:
Though in this case, it is not a repetition. In the regular expression, you want to match all variants of the field's name, and these variants are separated by a slash. So to match all of them in the regex, you need to transform the slash into a pipe.

The only issue here is that it makes the assumption that the second variant is for the plural form. A general solution should support variants which are just different spellings and variants which are plural forms, so that the regex can be built correnctly. Maybe we can discuss this in our next coding session.

Reply:
> Maybe we can discuss this in our next coding session.

Good idea!